### ISSUE TITLE
Auto-checking request body against Swagger JSON schema

### ISSUE THREAD

**Issue by mhelvens** [BigstickCarpet/swagger-express-middleware#16]
Since the Swagger spec supports JSON schemas for `req.body`, I expected the Validate Request middleware to use this to test incoming request bodies, and emit a (customizable) HTTP 4xx code on error. But that seems not to be the case?

If I am mistaken, please let me know how to enable this functionality. Otherwise, consider this a feature request.

In any case, thanks for these fantastic tools!

**Comment by BigstickCarpet** [comment]
This is actually done by the [Parse Request middleware](https://github.com/BigstickCarpet/swagger-express-middleware/blob/master/docs/middleware/parseRequest.md#parse-request-middleware), which is a [dependency](https://github.com/BigstickCarpet/swagger-express-middleware/blob/master/docs/middleware/validateRequest.md#dependencies) of the Validate Request middlware.   Here's the relevant documentation: [**Phase 2 - Swagger parsing**](https://github.com/BigstickCarpet/swagger-express-middleware/blob/master/docs/middleware/parseRequest.md#phase-2---swagger-parsing)

If `req.body` &mdash; or any other parameter in the request &mdash; is invalid, then an error will be thrown.  The HTTP error code varies, depending on the cause of the error.  Possible errors include 400, 411, and 500.   But, as with any error thrown by Swagger Express Middleware, you can catch the error in your own [error-handling middleware](http://expressjs.com/guide/error-handling.html) and change the error code to whatever you want.

**Comment by mhelvens** [comment]
I see. Thanks! I guess validation seemed like a Validation Middleware kind of thing. So much so that I didn't look further. I suppose there are reasons to keep this in the Parse Middleware. But perhaps a note in the Validation Middleware docs would be appropriate.

**Comment by BigstickCarpet** [comment]
Yeah, I can see how that would be confusing.  I'll re-open this issue to remind me to add that info to the docs.

**Comment by swissspidy** [comment]
I'm using both middlewares in the correct order in a project of mine.

While the required body param itself gets validated, the params it contains aren't. It seems like `parseBodyParam` and/or `parseParameter` need to check for these nested params too.

An excerpt from the schema:

```yaml
parameters:
  - name: data
    in: body
    required: true
    schema:
      properties:
        - name: name
          type: string
          required: true
        - name: email
          type: string
          required: true
```

Is this really not a bug?

**Comment by BigstickCarpet** [comment]
@swissspidy - Yes, the request body is validated against the schema in your Swagger API.  So, in your example, if the body doesn't contain a `name` property, then an HTTP 400 error will be thrown.  As @mhelvens pointed out, I need to document this better.

[Here's an example](https://github.com/BigstickCarpet/swagger-express-middleware/blob/master/docs/middleware/parseRequest.md#example) that demonstrates the functionality.

**Comment by samsep** [comment]
@BigstickCarpet as @swissspidy mentioned if I implement something like the above example
and fail to include an `email` property in the request body, the request still hits the controller and no errors are thrown. I'm also using both validator and parse middlewares.
